network-core/src/client/gossip.rs

```diff
@@ -1,15 +1,15 @@
 use super::Error;
 
-use super::super::gossip::{Gossip, NodeId};
+use super::super::gossip::{self, Gossip};
 
 use futures::prelude::*;
 
 pub trait GossipService {
     type Gossip: Gossip;
 
     /// Future that represent an asynchronous gossip request.
-    type GossipFuture: Future<Item = (NodeId, Self::Gossip), Error = Error>;
+    type GossipFuture: Future<Item = (gossip::Id, Self::Gossip), Error = Error>;
```

Comment:
I don't like the name `Id`, it's too short to mean anything. Could be confused with the ID of the gossip message by a casual reader.